src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/SourceControlTests.ps1

```diff
@@ -0,0 +1,322 @@
+# 2) Update the newly created source control - validate that is was updated
+# 3) Start a sync - validate that the sync completes 
+# 4) Retrieve the streams for the completed source control sync job
+# 5) Clean up - remove the source control
+
+# Automation account information
+$resourceGroupName = "frangom-test"
+$subscriptionId = "52d8cf1b-bcac-493a-bbae-f234b5ff38b0"
+$automationAccountName = "frangom-sdkCmdlet-tests"
+
+#region Helper functions
+
+# This function ensure that the given source control does not exist, 
+# which implicitly validates Get-AzureRmAutomationSourceControl and Remove-AzureRmAutomationSourceControl
+function EnsureSourceControlDoesNotExist
+{
+    Param 
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $Name
+    )
+
+    $sourceControl = Get-AzureRmAutomationSourceControl -ResourceGroupName $resourceGroupName `
+                                                        -AutomationAccountName $automationAccountName `
+                                                        -Name $Name `
+                                                        -ErrorAction SilentlyContinue
+    if ($sourceControl)
+    {
+        Remove-AzureRmAutomationSourceControl -ResourceGroupName $resourceGroupName `
+                                              -AutomationAccountName $automationAccountName `
+                                              -Name $Name
+    }
+}
+
+function WaitForSourceControlSyncJobState
+{
+    Param 
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $Name,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [Guid]
+        $JobId,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $ExpectedState
+    )
+
+    $state = ""
+    $waitTimeInSeconds = 5
+    $retries = 20
+
+    do
+    {
+        $syncJob = Get-AzureRmAutomationSourceControlSyncJob -ResourceGroupName $resourceGroupName `
+                                                             -AutomationAccountName $automationAccountName  `
+                                                             -Name $Name `
+                                                             -JobId $JobId
+
+        $state = $syncJob.ProvisioningState
+        Write-Output "Source control sync job Provisioning state: $state"
+
+        # Wait for 5 seconds and try again
+        sleep -Seconds 5
+        $retries--
+
+    } while ($state -ne $ExpectedState -and $retries -gt 0) 
+
+    Assert-True {$retries -gt 0} "Timeout waiting for provisioning state to reach '$ExpectedState'"
+}
+
+#endregion
+
+
+function Test-CreateVsoGitSourceControlAndSync
+{
+    # VsoGit repo info
+    $sourceControl = @{
+        Name = "AASourceControl-VsoGit"
+        RepoUrl = "https://francisco-gamino.visualstudio.com/_git/VsoGit-SwaggerAndCmdletsTests"
```

Comment:
?? This should either be configurable, or a public repo available to everyone

Reply:
Automation Source Control links your Automation account with their GitHub or VSTS (Azure Repos). This feature only works with Private repos.